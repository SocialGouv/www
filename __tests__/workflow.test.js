import child_process from "child_process";
import tempy from "tempy";
import util from "util";

const TIMEOUT = 1000 * 60;

const exec = util.promisify(child_process.exec);

test(
  "generate production k8s manifests",
  async () => {
    const dir = tempy.directory();
    console.log("dir", dir);
    const env = {
      GITHUB_JOB: "5678",
      GITHUB_REF: "refs/heads/mybranch",
      GITHUB_REPOSITORY: `socialgouv/www`,
      GITHUB_RUN_ID: "1234",
      GITHUB_SHA: "0123456789abcdefghijklmnopqrstuvwxyz0123",
      SOCIALGOUV_BASE_DOMAIN: "dev.fabrique.social.gouv.fr",
      SOCIALGOUV_CONFIG_PATH: __dirname + "/../.socialgouv/config.json",
      SOCIALGOUV_PREPRODUCTION: "",
      SOCIALGOUV_PRODUCTION: "true",
      SOCIALGOUV_PRODUCTION_NAMESPACE: "incubateur",
    };

    Object.assign(process.env, env);

    const cmd = `
      npx degit SocialGouv/kosko-charts/templates/autodevops ${dir}/autodevops; \
      yarn --cwd ${dir}/autodevops --silent; \
      cp -r ${__dirname}/../.socialgouv/environments ${dir}/autodevops/; \
      yarn --cwd ${dir}/autodevops --silent generate --env prod
    `;

    const { stdout: manifest } = await exec(cmd, { env: process.env });
    expect(manifest).toMatchSnapshot();
  },
  TIMEOUT
);
