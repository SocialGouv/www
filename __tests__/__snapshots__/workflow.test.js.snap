// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate production k8s manifests 1`] = `
"---
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/www
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    app: www
    application: www
    component: nginx
    owner: www
    team: www
  name: www
  namespace: incubateur
spec:
  replicas: 1
  selector:
    matchLabels:
      app: www
  template:
    metadata:
      annotations:
        kapp.k14s.io/disable-default-ownership-label-rules: ''
        kapp.k14s.io/disable-default-label-scoping-rules: ''
        app.github.com/job: '5678'
        app.github.com/ref: refs/heads/mybranch
        app.github.com/repo: socialgouv/www
        app.github.com/run: '1234'
        app.github.com/sha: '0123456'
      labels:
        app: www
        application: www
        component: nginx
        owner: www
        team: www
    spec:
      containers:
        - image: >-
            ghcr.io/socialgouv/fabrique/www:sha-0123456789abcdefghijklmnopqrstuvwxyz0123
          livenessProbe:
            failureThreshold: 6
            httpGet:
              path: /index.html
              port: http
            initialDelaySeconds: 30
            periodSeconds: 5
            timeoutSeconds: 5
          name: www
          ports:
            - containerPort: 80
              name: http
          readinessProbe:
            failureThreshold: 15
            httpGet:
              path: /index.html
              port: http
            initialDelaySeconds: 0
            periodSeconds: 5
            successThreshold: 1
            timeoutSeconds: 1
          resources:
            limits:
              cpu: 500m
              memory: 128Mi
            requests:
              cpu: 5m
              memory: 32Mi
          startupProbe:
            failureThreshold: 12
            httpGet:
              path: /index.html
              port: http
            periodSeconds: 5
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: www
    application: www
    component: nginx
    owner: www
    team: www
  name: www
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/www
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  namespace: incubateur
spec:
  ports:
    - name: http
      port: 80
      targetPort: 80
  selector:
    app: www
  type: ClusterIP
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
    cert-manager.io/cluster-issuer: letsencrypt-prod
    kubernetes.io/tls-acme: 'true'
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/www
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
    nginx.ingress.kubernetes.io/configuration-snippet: >-
      more_set_headers \\"Content-Security-Policy: default-src 'none'; connect-src
      'self' https://*.gouv.fr; font-src 'self'; img-src 'self'; prefetch-src
      'self' https://*.gouv.fr; script-src 'self' https://*.gouv.fr; frame-src
      'self' https://*.gouv.fr; style-src 'self' 'unsafe-inline'\\";

      more_set_headers \\"X-Frame-Options: deny\\";

      more_set_headers \\"X-XSS-Protection: 1; mode=block\\";

      more_set_headers \\"X-Content-Type-Options: nosniff\\";
  labels:
    app: www
    application: www
    component: nginx
    owner: www
    team: www
  name: www
  namespace: incubateur
spec:
  rules:
    - host: www.dev.fabrique.social.gouv.fr
      http:
        paths:
          - backend:
              service:
                name: www
                port:
                  name: http
            path: /
            pathType: Prefix
  tls:
    - hosts:
        - www.dev.fabrique.social.gouv.fr
      secretName: www-crt
---
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
    nginx.ingress.kubernetes.io/permanent-redirect: https://www.fabrique.social.gouv.fr$request_uri
    cert-manager.io/cluster-issuer: letsencrypt-prod
    kubernetes.io/tls-acme: 'true'
  labels:
    app.kubernetes.io/component: redirect
    app.kubernetes.io/name: www
  name: www-redirect
spec:
  rules:
    - host: www.incubateur.social.gouv.fr
    - host: incubateur.social.gouv.fr
    - host: fabrique.social.gouv.fr
  tls:
    - hosts:
        - www.incubateur.social.gouv.fr
        - incubateur.social.gouv.fr
        - fabrique.social.gouv.fr
      secretName: www-redirect
"
`;
